### ISSUE TITLE
struct declared inside @MainActor func can't be made Equatable

### ISSUE THREAD

**Issue by mayoff** [swiftlang/swift#60247]
This doesn't compile:

```swift
struct NonEquatable { }

@MainActor
func f() {
    struct Thing: Equatable {
        // ^ 🛑 Type 'Thing' does not conform to protocol 'Equatable'
        let prop: NonEquatable

        static func == (lhs: Thing, rhs: Thing) -> Bool { true }
    }
}
```

The compiler emits the error shown in the comment. The fix-it offered adds another declaration of `static func ==` with the same signature. The existing `static func ==` should satisfy the `Equatable` requirement.

I believe the compiler should accept this program. If there is a reason why it cannot accept the program, the compiler should emit the reason. Either it should explain why the given `static func ==` doesn't satisfy the `Equatable` requirement, or the fix-it should add a different declaration of `static func ==` that does satisfy the requirement.

**Environment (please fill out the following information)**

macOS 12.4 (21F79)
Xcode Version 14.0 beta 3 (14A5270f)

**Comment by AnthonyLatsis** [comment]
Duplicate of #45682

**Comment by AnthonyLatsis** [comment]
Fixed by #59246 on main, but not the release branch.
